src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,247 @@
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> models = new ArrayList<ViewModel>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      models.add(generate(model, service, apiConfig));
+    }
+    return models;
+  }
+
+  private StaticLangXCombinedView generate(Model model, Interface service, ApiConfig apiConfig) {
+    StaticLangXCombinedView.Builder view = StaticLangXCombinedView.newBuilder();
+
+    view.templateFileName(XAPI_TEMPLATE_FILENAME);
+    view.serviceDoc(comments(service));
+
+    GoSurfaceNamer namer = new GoSurfaceNamer(model, service, apiConfig);
+    SurfaceTransformerContext context =
+        SurfaceTransformerContext.create(service, apiConfig, createTypeTable(), namer);
+
+    view.packageName(namer.getPackageName());
+    view.clientName(namer.getClientName());
+    view.clientConstructorName(namer.getClientConstructorName());
+    view.defaultClientOptionFunc(namer.getDefaultClientOptionFunc());
+    view.defaultCallOptionFunc(namer.getDefaultCallOptionFunc());
+    view.callOptionsName(namer.getCallOptionsName());
+    view.serviceName(namer.getServiceName());
+    view.grpcClientTypeName(namer.getGrpcClientTypeName());
+    view.grpcClientConstructorName(namer.getGrpcClientConstructorName());
+    view.outputPath(namer.getOutputPath());
+
+    Collection<RetryPairDefinitionView> retryDef = generateRetryPairDefinitions(namer, context);
+    boolean hasRetry = !retryDef.isEmpty();
+    view.retryPairDefinitions(retryDef);
+
+    view.pathTemplates(pathTemplateTransformer.generatePathTemplates(context));
+    view.pathTemplateGetters(pathTemplateTransformer.generatePathTemplateGetterFunctions(context));
+    view.callSettings(apiCallableTransformer.generateCallSettings(context));
+    view.apiMethods(generateApiMethods(context));
+    view.pageStreamingDescriptorClasses(
+        pageStreamingTransformer.generateDescriptorClasses(context));
+
+    ServiceConfig serviceConfig = new ServiceConfig();
+    view.serviceAddress(serviceConfig.getServiceAddress(service));
+    view.servicePort(serviceConfig.getServicePort());
+    view.authScopes(serviceConfig.getAuthScopes(service));
+
+    view.imports(getImports(context, hasRetry));
+
+    return view.build();
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new GoTypeTable(), new GoModelTypeNameConverter());
+  }
+
+  /**
+   * Returns the doc comments of Go for the proto elements.
+   */
+  public static List<String> comments(ProtoElement element) {
+    if (!element.hasAttribute(ElementDocumentationAttribute.KEY)) {
+      return ImmutableList.<String>of("");
+    }
+    return new GoContextCommon().getCommentLines(DocumentationUtil.getScopedDescription(element));
```

Comment:
I would like to defer this until discovery samplegen migrates to MVMM. We cannot delete this class right now, and copying this functionality elsewhere just increases the chance of whoever migrating samplegen to MVVM to re-copy it.